src/message/MessageListWeb.js

```diff
@@ -72,17 +86,77 @@ export default class MessageListWeb extends Component<Props> {
       });
     }
 
-    if (renderedMessages !== nextProps.renderedMessages) {
-      const content = this.content(nextProps);
+    if (nextProps.updateMessages.length > this.props.updateMessages.length) {
+      // new messages are fetched & needs to be appended in the list
+      nextProps.updateMessages.forEach(messagesAction => {
+        const { action } = messagesAction;
+        // find where to append
+        if (action.numAfter === -1 && action.numBefore === -1) {
```

Comment:
We might have some new stuff here.
But we do not need to react per action type.
This is way too tedious and error-prone.